src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -624,7 +624,7 @@ private static TypeWithAnnotations DelegateReturnTypeWithAnnotations(MethodSymbo
                 returnType = inferredReturnType.TypeWithAnnotations;
                 returnRefKind = inferredReturnType.RefKind;
 
-                if (!returnType.HasType && returnTypes.Count > 0)
+                if (!returnType.HasType && inferredReturnType.NumExpressions > 0)
```

Comment:
>inferredReturnType.NumExpressions > 0 [](http://example.com/codeflow?start=43&length=37)

It looks like all added tests are success tests. Do we have failure tests for this condition? Specifically when there is a mix of returns with and without an expression.